src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointInvokerFactory.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    internal sealed class MvcEndpointInvokerFactory : IActionInvokerFactory
+    {
+        private readonly IActionInvokerFactory _invokerFactory;
+        private readonly IActionContextAccessor _actionContextAccessor;
+
+        public MvcEndpointInvokerFactory(
+            IActionInvokerFactory invokerFactory)
+            : this(invokerFactory, actionContextAccessor: null)
```

Comment:
How does this actually work? I thought we didn't have the ability to decorate services.

Reply:
idk. I copied it from here https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcAttributeRouteHandler.cs#L23-L47

Reply:
Ah, what I missed was how you register it https://github.com/aspnet/Mvc/pull/7886/files#diff-41b11542f8b97a386f2cce59031137bdR268